lib/beaker-puppet/wrappers.rb

```diff
@@ -45,7 +45,11 @@ def puppet(*args)
         options[:cmdexe] = true
         # we assume that an invocation with `puppet()` will have it's first argument
         # a face or sub command
-        cmd = "puppet #{args.shift}"
+        if @options[:type] == 'git'
+          cmd = "bundle exec puppet #{args.shift}"
+        else
+          cmd = "puppet #{args.shift}"
+        end
```

Comment:
So long as we don't install the runtime on the master we can skip this (as there won't be confusion about which puppet to run). Also the `puppet-agent` package won't clobber the untarred runtime.

Reply:
Should we not install the runtime on a host that's defined as both a master and an agent?